src/TrackedObjectBase.h

```diff
@@ -0,0 +1,115 @@
+/**
+ * @file
+ * @brief Header file for the TrackedObjectBase class
+ * @author Jonathan Thomas <jonathan@openshot.org>
+ *
+ * @ref License
+ */
+
+/* LICENSE
+ *
+ * Copyright (c) 2008-2019 OpenShot Studios, LLC
+ * <http://www.openshotstudios.com/>. This file is part of
+ * OpenShot Library (libopenshot), an open-source project dedicated to
+ * delivering high quality video editing and animation solutions to the
+ * world. For more information visit <http://www.openshot.org/>.
+ *
+ * OpenShot Library (libopenshot) is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * OpenShot Library (libopenshot) is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with OpenShot Library. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPENSHOT_TRACKEDOBJECTBASE_H
+#define OPENSHOT_TRACKEDOBJECTBASE_H
+
+#include <iostream>
+#include <iomanip>
+#include <cmath>
+#include <assert.h>
+#include <vector>
+#include <string>
+#include "Exceptions.h"
+#include "Fraction.h"
+#include "Coordinate.h"
+#include "KeyFrame.h"
+#include "Point.h"
+#include "Json.h"
+#include "ClipBase.h"
+
+
+namespace openshot {
+    /**
+	 * @brief This abstract class is the base class of all Tracked Objects.
+	 *
+	 * A Tracked Object is an object or a desired set of pixels in a digital image
+	 * which properties (such as position, width and height) can be detected and 
+	 * predicted along the frames of a clip.
+	 */
+    class TrackedObjectBase {
+	protected:
+		std::string id;
+		std::string childClipId;
+
+		ClipBase* parentClip;
+
+    public:
+		
+		Keyframe visible;
+		Keyframe draw_box;
+
+        /// Blank constructor
+        TrackedObjectBase();
+
+		/// Default constructor
+		TrackedObjectBase(std::string _id);
```

Comment:
```suggestion
		/// Default constructor
		TrackedObjectBase();

		/// Constructor which takes an object ID
		TrackedObjectBase(std::string _id);
```

The default construtor is one that can be called with no arguments. That is the definition